app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisController.kt

```diff
@@ -106,36 +108,55 @@ class VoiceAnalysisController @Inject constructor(
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribeBy(
-                onSuccess = { decibels -> handleNoiseDetectorData(decibels) },
+                onSuccess = { decibels ->
+                    handleNoiseDetectorData(
+                        decibels,
+                        recordingData.rawRecordingData
+                    )
+                },
                 onError = { error -> voiceAnalysisService?.complain("Noise detector error", error) }
             ).addTo(disposables)
     }
 
-    private fun handleNoiseDetectorData(decibels: Int) {
+    private fun handleNoiseDetectorData(decibels: Int, rawRecordingData: ByteArray) {
         debugModule.sendSoundEvent(decibels)
-        if (isNoiseDetected(decibels)) notifyBabyEventUseCase.notifyNoiseDetected()
+        if (isNoiseDetected(decibels)) {
+            feedbackController.handleRecording(rawRecordingData, false)
+                .subscribeOn(Schedulers.io())
+                .subscribeBy(
+                    onSuccess = { savedRecordingData ->
+                        Timber.i("Recording saved: ${savedRecordingData.fileName}" +
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/7c069c26782df05dac8a097aa956e351ad95f920/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisController.kt#L128"></span>Multiple occurrences of the same string literal within a single file detected.</td>
    </tr>
  </tbody>
</table>